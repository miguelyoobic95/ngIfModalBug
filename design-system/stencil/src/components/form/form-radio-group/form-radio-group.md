---
name: Radio Group
category: Form Components
---

## Attributes

|Name|Type|Default|Description|
|---|---|---|---|
|`text`|string|   |the text of the radio|
|`disabled`|boolean|false|whether the radio is disabled or not|
|`state`|string| |the starting state of the radio|

### Single Selection
```yoo-form-radio-group.html
    <yoo-form-radio-group class="accent"></yoo-form-radio-group>
```
```yoo-form-radio-group.js hidden
    var comp = document.querySelector('yoo-form-radio-group');
    comp.values = [{text: "Radio 1 Single Selection"}, {text: "Radio 2 Single Selection"},{text: "Radio 3 Single Selection"}];
```

### Multiple Selection
```yoo-form-radio-group-multiple.html
    <yoo-form-radio-group multiple-selection=true></yoo-form-radio-group>
```
```yoo-form-radio-group-multiple.js hidden
    var comp = document.querySelector('yoo-form-radio-group');
    comp.values = [{text: "Radio 1 Multiple Selection"}, {text: "Radio 2 Multiple Selection"},{text: "Radio 3 Multiple Selection"}];
```
## Events

|Attr|Description|
|---|---|---|
|`radioClicked`|event emitted when a radio state is changed|