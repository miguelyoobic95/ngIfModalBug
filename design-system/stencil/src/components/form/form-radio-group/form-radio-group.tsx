import { Component, Prop, Element, State, Event, EventEmitter } from '@stencil/core';
import { IRadioGroupEntry } from '@shared/interfaces';

@Component({
    tag: 'yoo-form-radio-group',
    styleUrl: 'form-radio-group.scss',
    scoped: true
})
export class YooFormRadioGroupComponent {

    @Prop() values: IRadioGroupEntry[] = [];
    @Prop() multipleSelection: boolean = false;

    @Event() selectionChanged: EventEmitter<Array<IRadioGroupEntry>>;

    @State() items: IRadioGroupEntry[] = [];

     @Element() host: HTMLStencilElement;

    private radioReset(): void {
        this.items = this.items.map((item, index) => {
            return {text: item.text, checked: false};
        });
    }

    private calculateRadioSelection(index: number) {
        if (this.multipleSelection) {
            this.items[index].checked = !this.items[index].checked;
        } else {
            let alreadyChecked = this.items[index].checked;
            this.radioReset();
            this.items[index].checked = !alreadyChecked;
        }
    }

    onRadioClicked(index: number) {
        this.calculateRadioSelection(index);
        this.selectionChanged.emit(this.items);
    }

    componentWillLoad() {
        this.items = this.values;
    }

    render(): JSX.Element {
        return (
            <div class="container" attr-layout="column">
                {this.items.map((obj, index) =>
                    <div class="inner-container">
                        <yoo-form-radio text={obj.text} class={this.host.className} state={obj.checked ? 'checked' : 'unchecked'} onRadioClicked={() => this.onRadioClicked(index)}></yoo-form-radio>
                    </div>
                )}
            </div>
        );
    }
}

