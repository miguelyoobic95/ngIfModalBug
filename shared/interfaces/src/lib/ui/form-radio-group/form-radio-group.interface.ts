export interface IRadioGroupEntry {
    text: string;
    checked?: boolean;
}