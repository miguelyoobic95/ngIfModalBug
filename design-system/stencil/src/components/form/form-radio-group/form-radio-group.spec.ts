import { TestWindow } from '@stencil/core/testing';
import { YooFormRadioGroupComponent } from './form-radio-group';

describe('YooFormRadioGroupComponent', () => {
    let window;
    beforeEach(() => {
        window = new TestWindow();
    });
    it('should build', () => {
        expect(new YooFormRadioGroupComponent()).toBeTruthy();
    });

    describe('Snapshots', () => {
        it('Should render', async () => {
            let element = await window.load({
                components: [YooFormRadioGroupComponent],
                html: '<yoo-form-radio-group class="accent"></yoo-form-radio-group>'
            });
            element.values = [{text: 'radio1'}, {text: 'radio2'}];
            await window.flush(element);
            expect(element).toMatchSnapshot();
           expect(element.values).toHaveLength(2);
        });

        it('Should render with multiple selection', async () => {
            let element = await window.load({
                components: [YooFormRadioGroupComponent],
                html: '<yoo-form-radio-group class="accent" multipleSelction=true></yoo-form-radio-group>'
            });
            element.values = [{text: 'radio1'}, {text: 'radio2'}];
            await window.flush(element);
            expect(element).toMatchSnapshot();
        });
    });

    describe('Methods', () => {
        describe('Single selection', () => {
            let element = new YooFormRadioGroupComponent();
            element.selectionChanged = {
                emit: () => {}
            };
            beforeEach(async () => {
                element.items = [{text: 'radio1'}, {text: 'radio2'}, {text: 'radio3'}, {text: 'radio4'}];
            });

            it('Should calculate correct item selection', async () => {
                const selectionChangedSpy = jest.spyOn(element.selectionChanged, 'emit');
                let result = [{text: 'radio1', checked: true}, {text: 'radio2', checked: false}, {text: 'radio3', checked: false}, {text: 'radio4', checked: false}];
                element.onRadioClicked(0);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result);
                expect(element.items).toEqual(result);
            });

            it('Should uncheck radio on second click', async () => {
                const selectionChangedSpy = jest.spyOn(element.selectionChanged, 'emit');
                let result1 = [{text: 'radio1', checked: true}, {text: 'radio2', checked: false}, {text: 'radio3', checked: false}, {text: 'radio4', checked: false}];
                let result2 = [{text: 'radio1', checked: false}, {text: 'radio2', checked: false}, {text: 'radio3', checked: false}, {text: 'radio4', checked: false}];
                element.onRadioClicked(0);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result1);
                expect(element.items).toEqual(result1);
                element.onRadioClicked(0);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result2);
                expect(element.items).toEqual(result2);
            });

            it('Should check correct radio on multiple selection', async () => {
                const selectionChangedSpy = jest.spyOn(element.selectionChanged, 'emit');
                let result1 = [{text: 'radio1', checked: true}, {text: 'radio2', checked: false}, {text: 'radio3', checked: false}, {text: 'radio4', checked: false}];
                let result2 = [{text: 'radio1', checked: false}, {text: 'radio2', checked: true}, {text: 'radio3', checked: false}, {text: 'radio4', checked: false}];
                let result3 = [{text: 'radio1', checked: false}, {text: 'radio2', checked: false}, {text: 'radio3', checked: true}, {text: 'radio4', checked: false}];
                let result4 = [{text: 'radio1', checked: false}, {text: 'radio2', checked: false}, {text: 'radio3', checked: false}, {text: 'radio4', checked: true}];
                element.onRadioClicked(0);
                expect(element.items).toEqual(result1);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result1);
                element.onRadioClicked(1);
                expect(element.items).toEqual(result2);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result2);
                element.onRadioClicked(2);
                expect(element.items).toEqual(result3);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result3);
                element.onRadioClicked(3);
                expect(element.items).toEqual(result4);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result4);
            });
        });

        describe('Multiple selection', () => {
            let element = new YooFormRadioGroupComponent();
            element.selectionChanged = {
                emit: () => {}
            };
            beforeEach(async () => {
                element.items = [{text: 'radio1'}, {text: 'radio2'}, {text: 'radio3'}, {text: 'radio4'}];
                element.multipleSelection = true;
            });
            it('Should check correct radio', async () => {
                const selectionChangedSpy = jest.spyOn(element.selectionChanged, 'emit');
                let result1 = [{text: 'radio1', checked: true}, {text: 'radio2'}, {text: 'radio3'}, {text: 'radio4'}];
                element.onRadioClicked(0);
                expect(element.items).toEqual(result1);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result1);
            });
            it('Should check and uncheck correct radio on multiple selection', async () => {
                const selectionChangedSpy = jest.spyOn(element.selectionChanged, 'emit');
                let result1 = [{text: 'radio1', checked: true}, {text: 'radio2'}, {text: 'radio3'}, {text: 'radio4'}];
                let result2 = [{text: 'radio1', checked: true}, {text: 'radio2', checked: true}, {text: 'radio3'}, {text: 'radio4'}];
                let result3 = [{text: 'radio1', checked: true}, {text: 'radio2', checked: true}, {text: 'radio3', checked: true}, {text: 'radio4'}];
                let result4 = [{text: 'radio1', checked: true}, {text: 'radio2', checked: true}, {text: 'radio3', checked: true}, {text: 'radio4', checked: true}];
                let result5 = [{text: 'radio1', checked: false}, {text: 'radio2', checked: true}, {text: 'radio3', checked: true}, {text: 'radio4', checked: true}];
                element.onRadioClicked(0);
                expect(element.items).toEqual(result1);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result1);
                element.onRadioClicked(1);
                expect(element.items).toEqual(result2);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result2);
                element.onRadioClicked(2);
                expect(element.items).toEqual(result3);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result3);
                element.onRadioClicked(3);
                expect(element.items).toEqual(result4);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result4);
                element.onRadioClicked(0);
                expect(element.items).toEqual(result5);
                expect(selectionChangedSpy).toHaveBeenCalledWith(result5);
            });
        });

    });
});
