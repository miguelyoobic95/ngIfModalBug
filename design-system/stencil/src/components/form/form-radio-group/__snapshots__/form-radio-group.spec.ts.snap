// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`YooFormRadioGroupComponent Snapshots Should render 1`] = `
<yoo-form-radio-group
  class="accent hydrated"
  data-yoo-form-radio-group-host=""
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-radio-group=""
  />
</yoo-form-radio-group>
`;

exports[`YooFormRadioGroupComponent Snapshots Should render with multiple selection 1`] = `
<yoo-form-radio-group
  class="accent hydrated"
  data-yoo-form-radio-group-host=""
  multipleselction="true"
>
  
  <div
    attr-layout="column"
    class="container"
    data-yoo-form-radio-group=""
  />
</yoo-form-radio-group>
`;
